src/test/scala/io/iohk/ethereum/consensus/validators/BlockWithCheckpointHeaderValidatorSpec.scala

```diff
@@ -0,0 +1,341 @@
+    val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent)
+    assert(validateResult == expectedError)
+    val blockHeader2 = validBlockHeaderWithCheckpoint.copy(treasuryOptOut = Some(true))
+    val validateResult2 = blockHeaderValidator.validate(blockHeader2, validBlockParent)
+    assert(validateResult2 == expectedError)
+  }
+
+  it should "return failure if stateRoot is not the same as parent stateRoot" in new TestSetup {
+    testOfTheSameValueAsParent(
+      byteString => validBlockHeaderWithCheckpoint.copy(stateRoot = byteString),
+      "stateRoot",
+      validBlockParent.stateRoot
+    )
+  }
+
+  it should "return failure if created based on invalid timestamp" in new TestSetup {
+    forAll(longGen suchThat (_ != validBlockParent.unixTimestamp + 1)) { timestamp =>
+      val blockHeader = validBlockHeaderWithCheckpoint.copy(unixTimestamp = timestamp)
+      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent)
+      assert(validateResult == Left(HeaderTimestampError))
+    }
+  }
+
+  it should "return failure if difficulty is different than parent difficulty" in new TestSetup {
+    forAll(bigIntGen suchThat (_ != validBlockParent.difficulty)) { difficulty =>
+      val blockHeader = validBlockHeaderWithCheckpoint.copy(difficulty = difficulty)
+      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent)
+      assert(validateResult == Left(HeaderNotMatchParentError("difficulty has different value that similar parent field")))
+    }
+  }
+
+  it should "return failure if gas used is not zero" in new TestSetup {
+    forAll(bigIntGen suchThat (_ != UInt256.Zero.toBigInt)) { gasUsed =>
+      val blockHeader = validBlockHeaderWithCheckpoint.copy(gasUsed = gasUsed)
+      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent)
+      assert(validateResult == Left(HeaderGasUsedError))
+    }
+  }
+
+  it should "return failure if gas limit is different than parent gas limit" in new TestSetup {
+    forAll(bigIntGen suchThat (_ != validBlockParent.gasLimit)) { gasLimit =>
+      val blockHeader = validBlockHeaderWithCheckpoint.copy(gasLimit = gasLimit)
+      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent)
+      assert(validateResult == Left(HeaderNotMatchParentError("gasLimit has different value that similar parent field")))
+    }
+  }
+
+  it should "return failure if created based on invalid number" in new TestSetup {
+    forAll(longGen suchThat (num => num != validBlockParent.number + 1 && num >= config.ecip1097BlockNumber)) { number =>
+      val blockHeader = validBlockHeaderWithCheckpoint.copy(number = number)
+      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent)
+      assert(validateResult == Left(HeaderNumberError) || validateResult == Left(HeaderDifficultyError))
```

Comment:
>validateResult == Left(HeaderDifficultyError)

Shouldn't it never be this case? The blockHeader has a checkpoint so the validation that would result in this error is never done